src/main/java/com/worksap/nlp/sudachi/DictionaryFactory.java

```diff
@@ -104,10 +106,22 @@ public Dictionary create(String path, String settings) throws IOException {
      */
     @Deprecated
     public Dictionary create(String path, String settings, boolean mergeSettings) throws IOException {
-        Config config = Config.fromSettings(Settings.parseSettings(path, settings));
-        if (mergeSettings) {
-            config = config.withFallback(Config.fromClasspath());
+        Config config;
+        SettingsAnchor defaultAnchor = SettingsAnchor.classpath();
+        if (path != null) {
+            defaultAnchor = SettingsAnchor.filesystem(Paths.get(path)).andThen(defaultAnchor);
```

Comment:
In the original logic, `path` does not specify where sudachi.json is to be loaded (it just overrides the path attribute of sudachi.json). Would you like to make it compatible with 0.5.3?
https://github.com/WorksApplications/Sudachi/blob/df81119a1ca87644b33eef4987d1a60edacc1d67/src/main/java/com/worksap/nlp/sudachi/JapaneseDictionary.java#L89

Reply:
Anchor does not mean that sudachi.json will be loaded from that location, it means that paths inside sudachi.json will be resolved with respect to that location. In this case it is the passed path, then search in classpath, which is the original behavior as far as I understand it.